go-controller/pkg/util/subnet_annotations.go

```diff
@@ -37,82 +38,109 @@ const (
 	ovnNodeSubnets = "k8s.ovn.org/node-subnets"
 )
 
-func createSubnetAnnotation(annotationName string, defaultSubnets []*net.IPNet) (map[string]interface{}, error) {
+// updateSubnetAnnotation add the hostSubnets of the given network to the input node annotations;
+// if hostSubnets is empty, it delete the existing subnet annotation for given network from the input node annotations.
+func updateSubnetAnnotation(annotations map[string]string, annotationName, netName string, hostSubnets []*net.IPNet) error {
 	var bytes []byte
 	var err error
 
-	if len(defaultSubnets) == 1 {
-		bytes, err = json.Marshal(map[string]string{
-			"default": defaultSubnets[0].String(),
-		})
+	// First get the all host subnets for all existing networks
+	subnetsMap := map[string][]*net.IPNet{}
+	_, ok := annotations[annotationName]
+	if ok {
+		subnetsMap, err = parseSubnetAnnotation(annotations, annotationName)
+		if err != nil {
+			return fmt.Errorf("failed to parse node subnet annotation %q: %v",
+				annotations, err)
+		}
+	}
```

Comment:
maybe export to an helper func.

Reply:
changed the parseSubnetAnnotation() function, please take a look